core/src/test/java/org/apache/iceberg/view/TestViewMetadata.java

```diff
@@ -0,0 +1,280 @@
+    assertThatThrownBy(() -> ImmutableViewMetadata.builder().build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [formatVersion, location, currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(() -> ImmutableViewMetadata.builder().formatVersion(1).build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [location, currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(
+            () -> ImmutableViewMetadata.builder().formatVersion(1).location("location").build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [currentVersionId]");
+
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid view versions: empty");
+  }
+
+  @Test
+  public void unsupportedFormatVersion() {
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(23)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Unsupported format version: 23");
+  }
+
+  @Test
+  public void emptyViewVersion() {
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid view versions: empty");
+  }
+
+  @Test
+  public void emptySchemas() {
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .addVersions(
```

Comment:
Does this method accept multiple versions or should the name be singular rather than plural?

Reply:
there's a `addVersions(ViewVersion element)` and a `addVersions(ViewVersion... elements)` but no `addVersion(ViewVersion element)` because the name is being derived from the naming in the API, which is `versions()`

Reply:
I guess that's okay.